src/Writer.php

```diff
@@ -11,28 +14,12 @@ class Writer implements WriterInterface
      * @see \Hostnet\Component\EntityPlugin\WriterInterface::writeFile()
      * @param string $path File path including directory
      * @param string $data File contents
+     * @throws \Symfony\Component\Filesystem\Exception\IOException when directory can not be created or
+     *         file is unwriteable
      */
     public function writeFile($path, $data)
     {
-        $dir = dirname($path);
-        $this->ensureDirectoryExists($dir);
-
-        if (! is_dir($dir)) {
-            mkdir($path, 0755, true);
-        }
-        file_put_contents($path, $data);
-    }
-
-    /**
-     * Ensures that the Generated/ folder exists
-     *
-     * @throws \RuntimeException
-     * @param string $path Directory Path
-     */
-    private function ensureDirectoryExists($path)
-    {
-        if (! is_dir($path) && ! mkdir($path)) {
-            throw new \RuntimeException('Could not create "Generated" directory "' . $path . '"');
-        }
+        $fs = new Filesystem();
```

Comment:
Good find, this saves a lot of code :+1: 

Would be even more awesome if this was created in the `Plugin::activate` as well. Though this is already an improvement, so no blocking remark.